modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java

```diff
@@ -695,16 +696,36 @@ private int _dynamicSearchCount(
 		return availableClassTypeIds;
 	}
 
-	private long[] _getCombinedSegmentsEntryIds(long[] segmentEntryIds) {
+	private long[] _getCombinedSegmentsEntryIds(
+		AssetListEntry assetListEntry, long[] segmentEntryIds) {
+
 		if ((segmentEntryIds.length > 1) &&
 			ArrayUtil.contains(
 				segmentEntryIds, SegmentsEntryConstants.ID_DEFAULT)) {
 
-			return ArrayUtil.remove(
+			segmentEntryIds = ArrayUtil.remove(
```

Comment:
could we make this list inmutable someway?

Reply:
The array returned by ArrayUtil.remove(...) is already immutable as it's created specifying the length of it.

Internally, that remove treats the given array as a List and later on transform it again to an array with the following method:

`public static long[] toArray(Long[] array) {
		long[] newArray = new long[array.length];

		for (int i = 0; i < array.length; i++) {
			newArray[i] = array[i].longValue();
		}

		return newArray;
	}`

Reply:
Hey, but we have muted segmentEntryIds, that is a parameter, right?, or I am not getting something right?